iterative/utils/analytics.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"regexp"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shirou/gopsutil/host"
+	"github.com/sirupsen/logrus"
+)
+
+func SystemInfo() map[string]interface{} {
+	hostStat, _ := host.Info()
+	return map[string]interface{}{
+		"os":               hostStat.OS,
+		"platform":         hostStat.Platform,
+		"platform_version": hostStat.PlatformVersion,
+	}
+}
+
+func TaskDuration(logs string) float64 {
+	regex := regexp.MustCompile(`\w{3} \d{2} \d{2}:\d{2}:\d{2}`)
+	matches := regex.FindAllString(logs, -1)
+	taskDuration := 0.0
+
+	if len(matches) > 1 {
+		layout := "Mar 07 06:43:27"
+		t1, _ := time.Parse(layout, matches[len(matches)-1])
+		t0, _ := time.Parse(layout, matches[0])
+		taskDuration = t1.Sub(t0).Seconds()
+	}
+
+	return taskDuration
+}
+
+func IsCI() bool {
+	if _, ok := os.LookupEnv("CI"); ok {
+		return true
+	}
+
+	return false
+}
+
+func version() string {
+	return "v1.0.0"
+}
+
+func TerraformVersion() string {
+	var out bytes.Buffer
+	cmd := exec.Command("terraform", "--version")
+	cmd.Stdout = &out
+
+	err := cmd.Run()
+
+	if err == nil {
+		regex := regexp.MustCompile(`v\d+\.\d+\.\d+`)
+		version := regex.FindString(out.String())
+		return version
+	} else {
+		logrus.Error("Analytics: Failed extracting terraform version")
+	}
+
+	return ""
+}
+
+func JitsuUserId() string {
+	hostStat, _ := host.Info()
+	id := hostStat.HostID
+
+	if IsCI() {
+		id = fmt.Sprintf("%s%s%s",
+			os.Getenv("GITHUB_REPOSITORY_OWNER"),
+			os.Getenv("CI_PROJECT_ROOT_NAMESPACE"),
+			os.Getenv("BITBUCKET_WORKSPACE"),
+		)
+	}
```

Comment:
According to the specs we should use multiple vars to even gather the userid. I think that with the organization it suffices. Not sure why we want to dig into specific users within the organization

Reply:
The other variables are needed for [`user_id`](https://www.notion.so/iterative/Specification-1a0197b8e34d43a89777a4c9a00ad1fe#7f4e4247c0ef4331a9243cbdb3792b18) but what you have is `group_id` more or less. 🤔 Please take another look to [this](https://www.notion.so/iterative/Specification-1a0197b8e34d43a89777a4c9a00ad1fe#c4a1c6a1e0b94d0d972e5e728fcc02d8).

Reply:
Thats what I say 

`GITHUB_REPOSITORY_OWNER ` is the GH organization or the user
`CI_PROJECT_ROOT_NAMESPACE` mean group/subgroup/subgroup if any
`BITBUCKET_WORKSPACE` is like GL's one

Reply:
What do we understand for user_id?

Reply:
`user_id` is `anonymize('DavidGOrtega <g···d@gmail.com>')`
`group_id` is `anonymize('https://github.com/iterative')`

Reply:
thats what I say. 
I should only need to have `iterative`, I do not care about `DavidGOrtega`

Reply:
I find both useful, because we can tell apart the number of contributors (e.g. size of an organization)

Reply:
This sounds like a good topic to comment [on Notion](https://www.notion.so/iterative/Specification-1a0197b8e34d43a89777a4c9a00ad1fe#7f4e4247c0ef4331a9243cbdb3792b18)

Reply:
> because we can tell apart the number of contributors (e.g. size of an organization)

How er can tell? The id is anonymised

Reply:
Count unique `user_id` values for a given `group_id` ❓